hera_cal/firstcal.py

```diff
@@ -403,7 +403,7 @@ def UVData_to_dict(uvdata_list, filetype='miriad'):
         flags = uv_in.flag_array.reshape(
             uv_in.Ntimes, uv_in.Nbls, uv_in.Nspws, uv_in.Nfreqs, uv_in.Npols)
 
-        for nbl, (i, j) in enumerate(map(uv_in.baseline_to_antnums, uv_in.baseline_array[:uv_in.Nbls])):
+        for nbl, (i, j) in enumerate(map(uv_in.baseline_to_antnums, np.unique(uv_in.baseline_array))):
```

Comment:
np.unique returns a sorted list of unique elements. Are the first Nbls elements necessarily sorted in the baseline array?  I can see this being a problem in the loop since data is being indexed by nbl. If the bl order in the data doesn't correspond to the sorted baseline array of the loop, the dictionaries would have the wrong mapping.

Reply:
yeah good point, let me look into that..

Reply:
We can also get around this by just using the get_data methods for a uvdata object.

Reply:
ok, good point, ill try that

Reply:
i think that worked well!